bson/doc.go

```diff
@@ -39,4 +39,89 @@
 // 		err = bson.Unmarshal(b, &fooer)
 // 		if err != nil { return err }
 // 		// do something with fooer...
+//
+// The default encoding/decoding mappings for Go to BSON types are:
```

Comment:
This is good information to document, but it doesn't seem to fit quite right at this location.  Some of the content seems to only apply to the struct codec (e.g. "Strings can be decoded from a BSON string, ObjectId, or symbol."), yet struct encoding/decoding is not mentioned above.  Most of it applies to `bson.M`, but `bson.M` is not (yet, at least) mentioned above, though `bson.D` is.

Also, for the supported dynamic types, `bson.M` and `bson.D`, it's important to document the default Go type that each BSON type decodes into. e.g that a BSON string decodes by default to a Go string, a BSON Int64 decodes into ..., etc.

Reply:
Agreed that this isn't quite right. I'll schedule some time for us to talk about this in person and figure out where things need to go. I like the idea about putting some of this on the `StructCodec` documentation, but I don't think users are likely to find that by themselves, so there needs to be some top-level docs, even if they just act as a table of contents to tell users where to go.